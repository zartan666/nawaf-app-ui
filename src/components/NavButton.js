import React from 'react';
import {StyleSheet, Text, TouchableHighlight} from 'react-native';
import colors from './../common/colors';
import Ionicons from 'react-native-vector-icons/Ionicons';

export default class NavButton extends React.Component {
  static propTypes = {
    // style:View.propTypes.style,
    text: React.PropTypes.string,
    icon: React.PropTypes.oneOfType([
      React.PropTypes.string,
      React.PropTypes.element,
    ]),
    onPress: React.PropTypes.func.isRequired,
  };

  handlePress() {
    this.props.onPress();
  }

  render() {
    const {style, title, icon, iconSize} = this.props;

    return (
      <TouchableHighlight
        onPress={() => {
          this.handlePress();
        }}
        underlayColor="transparent"
        activeOpacity={0.4}
        style={styles.container}>
        {icon
          ? typeof icon == 'string'
              ? <Ionicons
                  name={icon}
                  size={iconSize ? iconSize : 40}
                  color={colors.accent}
                  style={[styles.icon, style]}
                />
              : icon
          : <Text style={[styles.title, style]}>{title}</Text>}
      </TouchableHighlight>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    flexDirection: 'row',
    alignItems: 'center',
    paddingLeft: 15,
    paddingRight: 15,
    // backgroundColor:'black',
  },
  title: {
    color: colors.accent,
    fontSize: 15,
  },
  icon: {
    width: 40,

    height: 40,
    alignSelf: 'center',
    color: colors.accent,
  },
});
