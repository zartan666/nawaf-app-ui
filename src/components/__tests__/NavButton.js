import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavButton from './../NavButton';

test('renders null', () => {
  const tree = renderer
    .create(<NavButton onPress={() => {}} text="title" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
