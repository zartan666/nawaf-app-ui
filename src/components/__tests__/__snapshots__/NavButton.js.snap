// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders null 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingLeft": 15,
        "paddingRight": 15,
      },
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#ff1144",
          "fontSize": 15,
        },
        undefined,
      ]
    }
  />
</View>
`;
